tests/commons/src/main/java/org/codice/ddf/admin/comp/test/AbstractComponentTest.java

```diff
@@ -0,0 +1,177 @@
+                .flatMap(Collection::stream)
+                .toArray(Option[]::new);
+    }
+
+    public List<Option> customSettings() {
+        return new ArrayList<>();
+    }
+
+    public List<Option> distributionSettings() {
+        return Arrays.asList(debugConfiguration("50005", Boolean.getBoolean("isDebugEnabled")),
+                karafDistributionConfiguration().frameworkUrl(maven().groupId("org.codice.ddf")
+                        .artifactId("kernel")
+                        .versionAsInProject()
+                        .type("zip"))
+                        .unpackDirectory(new File("target/exam"))
+                        .useDeployFolder(false),
+                cleanCaches());
+    }
+
+    public List<Option> configurableSettings() throws Exception {
+        return Arrays.asList(keepRuntimeFolder(),
+                logLevel(LogLevelOption.LogLevel.INFO),
+
+                // TODO: tbatie - 8/15/17 - Change these to dynamic port once refactored
+                editConfigurationFilePut("etc/system.properties", "ddf.home", "${karaf.home}"),
+                editConfigurationFilePut("etc/system.properties",
+                        "org.codice.ddf.system.httpsPort",
+                        "9993"),
+                editConfigurationFilePut("etc/system.properties",
+                        "org.codice.ddf.system.httpPort",
+                        "9994"),
+                editConfigurationFilePut("etc/system.properties",
+                        "org.codice.ddf.catalog.ftp.port",
+                        "9995"),
+                editConfigurationFilePut("etc/org.apache.karaf.management.cfg",
+                        "rmiRegistryPort",
+                        "20001"),
+                editConfigurationFilePut("etc/org.apache.karaf.management.cfg",
+                        "rmiServerPort",
+                        "20002"),
+                editConfigurationFilePut("etc/org.apache.karaf.management.cfg",
+                        "rmiServerPort",
+                        "20002"),
+                editConfigurationFilePut("etc/org.apache.karaf.features.cfg",
+                        "featuresBootAsynchronous",
+                        "true"));
+    }
+
+    @BeforeExam
+    public void beforeExam() throws Exception {
+        adminConfig = new AdminConfig(configurationAdmin);
+        serviceManager = (ServiceManager) Proxy.newProxyInstance(getClass().getClassLoader(),
+                ServiceManagerImpl.class.getInterfaces(),
+                new ServiceManagerProxy(new ServiceManagerImpl(metatypeService, adminConfig)));
+
+        securityPolicyConfigurator = new SecurityPolicyConfigurator(new ServiceManagerImpl(
+                metatypeService,
+                adminConfig), configurationAdmin);
+        startFeatures();
+    }
+
+    protected List<Option> configureVmOptions() {
+        return Arrays.asList(vmOption("-Xmx4096M"),
```

Comment:
✏️ If you add a second `vmOption` for the minheap (`-Xms`) and set it to at least `2G`/`2048M`, you'll probably see an improvement in startup and overall runtime for the configuration tests. If you set it initially to 4G, you'd get no memory thrashing at all, but that's probably more space than you need and there are costs to allocating the heap.